lms/djangoapps/grades/signals.py

```diff
@@ -124,19 +125,33 @@ def recalculate_subsection_grade_handler(sender, **kwargs):  # pylint: disable=u
     """
     if not settings.FEATURES.get('ENABLE_SUBSECTION_GRADES_SAVED', False):
         return
-    try:
-        course_id = kwargs.get('course_id', None)
-        usage_id = kwargs.get('usage_id', None)
-        student = kwargs.get('user', None)
 
-        course_key = CourseLocator.from_string(course_id)
-        usage_key = UsageKey.from_string(usage_id).replace(course_key=course_key)
+    from lms.djangoapps.grades.transformer import GradesTransformer
+    from lms.djangoapps.grades.new.subsection_grade import SubsectionGradeFactory
+    from openedx.core.djangoapps.content.block_structure.api import get_block_structure_manager
```

Comment:
I believe some of these `import`s can be included at the top of the file instead of here, I'm unrolling the circular imports now.